### PR TITLE
chore: Convert proptests to test strategy

### PR DESCRIPTION
**Pull request by berestovskyy**
There is no change in functionality. The benefit of using `#[strategy]` annotations is that the code is formatted as expected, unlike the code within the `proptest! {...}` macro.

### CODE REVIEW

File: rs/replicated_state/src/canister_state/queues/tests.rs [hunk 1]
```diff
@@ -1950,23 +1950,27 @@ prop_compose! {
...
+    while let Some(msg) = output_iter.peek() {
+        popped += 1;
+        prop_assert_eq!(Some(msg.clone()), output_iter.next());
     }
 
+    prop_assert_eq!(output_iter.next(), None);
+    prop_assert_eq!(raw_requests.len(), popped);
+}
+
+proptest! {
```
Comment (schneiderstefan) [inline rs/replicated_state/src/canister_state/queues/tests.rs:1973]: Why stop here and go back to the old macro?

Comment (berestovskyy) [reply rs/replicated_state/src/canister_state/queues/tests.rs:1973]: I put in the description:

> The followup PRs will convert the rest of the tests.

It's just a start, we'll see how it goes.

**Comment by ielashi** [comment]
@berestovskyy Thank you! I really wanted to see us make this change at some point.

**Review by ielashi — APPROVED** [review]

**Review by schneiderstefan — APPROVED** [review]

File: rs/replicated_state/src/canister_state/queues/tests.rs [hunk 2]
```diff
@@ -1950,23 +1950,27 @@ prop_compose! {
...
-        prop_assert_eq!(output_iter.next(), None);
-        prop_assert_eq!(raw_requests.len(), popped);
+#[test_strategy::proptest]
+fn output_into_iter_peek_and_next_consistent(
+    #[strategy(arb_canister_output_queues(10, Some(5)))] test: (
+        CanisterQueues,
+        VecDeque<RequestOrResponse>,
+    ),
+) {
+    let (mut canister_queues, raw_requests) = test;
```
Comment (alin-at-dfinity) [inline rs/replicated_state/src/canister_state/queues/tests.rs:1960]: Mostly just curious: couldn't this have been written as

```suggestion
    #[strategy(arb_canister_output_queues(10, Some(5)))] (mut canister_queues, raw_requests): (
        CanisterQueues,
        VecDeque<RequestOrResponse>,
    ),
) {
```

**Review by alin-at-dfinity — APPROVED** [review]

**Comment by berestovskyy** [comment]
FYI folks, I've converted the remaining tests. The diff looks messy, so I hope I didn't mix up the arguments...

**Pull request merged**